src/ffi.rs

```diff
@@ -429,3 +412,16 @@ pub fn i2c_smbus_process_call_block(fd: RawFd, register: u8, values: &[u8]) -> R
     let count = data.block[0];
     Ok((&data.block[1..(count + 1) as usize]).to_vec())
 }
+
+#[inline]
+pub fn i2c_rdwr(fd: RawFd, values: &mut [i2c_msg]) -> Result<u32, I2CError> {
+    let i2c_data = i2c_rdwr_ioctl_data {
+        msgs: values.as_mut_ptr(),
+        nmsgs: values.len() as u32,
+    };
+
+    unsafe {
+        let n = ioctl::i2c_rdwr(fd, &i2c_data)?;
+        Ok(n as u32)
```

Comment:
Nit: move safe portions of code out of unsafe block.

Reply:
done.